### ISSUE TITLE
crnn训练部分 lable问题

### ISSUE THREAD

**Issue by qixxtny** [ooooverflow/chinese-ocr#12]
def __getitem__(self, index):
		label = [int(x) for x in keys] 
        请问label是要识别的str，怎么转成int呢？

**Comment by guess3w** [comment]
同问，这个怎么解决？用字典编号解决？
看了下代码，反推识别的字是如何得来的:result->cor.model()->carRec()->predict()->decode()
在orc.py中 decode()函数里，貌似是用char_set里的index来的

**Comment by Miscoding** [comment]
> 同问，这个怎么解决？用字典编号解决？
> 看了下代码，反推识别的字是如何得来的:result->cor.model()->carRec()->predict()->decode()
> 在orc.py中 decode()函数里，貌似是用char_set里的index来的

data_train.txt 里应该就是 图像名+每个汉字的对应字典索引数字 但是 decode（）函数中
`if preds[i] != 0 and ((i == 0) or (i != 0 and preds[i] != preds[i-1])):
pred_text += char_set[int(preds[i])-1]`
为什么要preds[i]!=0 并且char_set预测结果要index-1呢？label中每个汉字的对应索引要加1吗
请问你能完整训练一遍代码吗？

**Comment by guess3w** [comment]
你看一下那个charset.txt第一行是空行，所以要index减1。
我可以训练了，但是训练一段时间，会出现数据维度错误，导致训练终止，估计是我下载的那个大的样本数据库，解压时产生错误。周一上班再调一下。
 
---原始邮件---
发件人: "Miscoding"<notifications@github.com&gt;
发送时间: 2019年9月1日(星期日) 晚上9:36
收件人: "ooooverflow/chinese-ocr"<chinese-ocr@noreply.github.com&gt;;
抄送: "guess3w"<77351470@qq.com&gt;;"Comment"<comment@noreply.github.com&gt;;
主题: Re: [ooooverflow/chinese-ocr] crnn训练部分 lable问题 (#12)


 
同问，这个怎么解决？用字典编号解决？
 看了下代码，反推识别的字是如何得来的:result-&gt;cor.model()-&gt;carRec()-&gt;predict()-&gt;decode()
 在orc.py中 decode()函数里，貌似是用char_set里的index来的
  
data_train.txt 里应该就是 图像名+每个汉字的对应字典索引数字 但是 decode（）函数中
 if preds[i] != 0 and ((i == 0) or (i != 0 and preds[i] != preds[i-1])): pred_text += char_set[int(preds[i])-1]
 为什么要preds[i]!=0 并且char_set预测结果要index-1呢？label中每个汉字的对应索引要加1吗
 请问你能完整训练一遍代码吗？

**Comment by guess3w** [comment]
@Miscoding preds[i]!=0 好像是ctc loss里的空白占位符,我也是刚看。

**Comment by guess3w** [comment]
@Miscoding if preds[i] != 0 and ((i == 0) or (i != 0 and preds[i] != preds[i-1]))这个是CTC LOSS里的，两个连续的输出值一样的话视为同一字符，除非用占位符0分隔开。

**Comment by Miscoding** [comment]
@guess3w 那训练的时候label 中汉字的字典索引是不是要从1开始呢？

**Comment by guess3w** [comment]
@Miscoding 个人理解不用，我用的是label = [char_set.index(x) for x in list(keys[0])]
char_set 已经将第一个去掉了(char_set[1:])

**Comment by guess3w** [comment]
@Miscoding 训练时，需要将train_loader那里加一个参数，drop_last=true，不然样本数不能整除batch_size时，会出错。。

**Comment by zblfly** [comment]
> @Miscoding 个人理解不用，我用的是label = [char_set.index(x) for x in list(keys[0])] char_set 已经将第一个去掉了(char_set[1:])

博主可以请教下是怎么训练的吗？我按您的改了以后还是无法成功训练
